src/pages/Dashboard.tsx

```diff
@@ -53,23 +59,42 @@ export default () => {
     setTransactionResult
   ] = useState<PagedResult<Transaction>>({ rows: [], count: 0 });
 
+  let infoSubscription: Subscription;
+
   useEffect(() => {
-    setBlocksPending(true);
-    fetchBlocks({ startAt: 0, endAt: blocksPerPage }).then(payload => {
-      setBlocksPending(false);
-      setBlockResult(payload);
-    });
-  }, [blocksPerPage]);
+    if (!infoSubscription) {
+      infoSubscription = observeInfo().subscribe(info => {
+        latestInfo.update(info);
+        setSnapshotHeight(info.blockHeight);
+      });
+    }
+
+    return function cleanup() {
+      infoSubscription.unsubscribe();
```

Comment:
socket connection needs to be removed on page exit

Reply:
When you're saying that react isn't yours, apparently you kinda know what you're talking about... ;-)